src/app/planning-room/planning-room.component.ts

```diff
@@ -35,11 +38,18 @@ export class PlanningRoomComponent implements OnInit, OnDestroy {
       .subscribe(params => this.userEmail = params['userEmail'])
     );
 
+    // Get key and userEmail
+    this.key = this.route.snapshot.paramMap.get('key');
+
     // Subscribe to planning
     this.subsctiptions$.push(this.route.params
```

Comment:
Aca si ya tenes el valor de la key en this.key no necesitas hacer uso de this.route.params.pipe(concatMap)
podes omitir eso y directamente llamas al observable this.planningSessionService.getPlanningByKey(this.key))

Reply:
Done.